vit-servicing-station-tests/src/tests/rest/funds.rs

```diff
@@ -39,9 +41,7 @@ pub fn get_funds_by_id() -> Result<(), Box<dyn std::error::Error>> {
 
     let mut actual_fund = rest_client.fund(&expected_fund.id.to_string())?;
 
-    actual_fund.chain_vote_plans.sort_by_key(|c| c.id);
-    expected_fund.challenges.sort_by_key(|c| c.internal_id);
-    expected_fund.chain_vote_plans.sort_by_key(|c| c.id);
+    normalize(&mut actual_fund);
```

Comment:
Maybe I missed something, but I have thought that at this point we expect to get already well structured fund. Am I correct with my assumption ?

Reply:
The problem is that `challenge.internal_id` is set by the test, but ignored when putting it in the database. Instead, the database uses its own auto-incrementing integer.

So if we generate a fund with, for example:
```rust
challenges: vec![
  Challenge {
    internal_id: 2,
   // other fields
  },
  Challenge {
    internal_id: 1,
  },
]
```
this will be put into the database with the `internal_id` fields swapped.

So we call `normalize` before putting it into the database and after retrieving it, to make sure we're always dealing with data in the same order

Reply:
Ideally, we'd have 2 structs like:
```rust
#[derive(Queryable)]
struct Challenge {
  internal_id: i32,
  // other fields,
}

#[derive(Insertable)] 
struct InsertChallenge {
  // no internal_id field
  // other fields
}
```
That way, we *can't* specify a meaningless `internal_id`, but I suspect this would be a pretty large refactor